error/s2n_errno.h

```diff
@@ -165,3 +166,5 @@ extern __thread const char *s2n_debug_str;
 #define _S2N_ERROR( x )     do { s2n_debug_str = _S2N_DEBUG_LINE; s2n_errno = ( x ); } while (0)
 #define S2N_ERROR( x )      do { _S2N_ERROR( ( x ) ); return -1; } while (0)
 #define S2N_ERROR_PTR( x )  do { _S2N_ERROR( ( x ) ); return NULL; } while (0)
+#define S2N_ERROR_IF( cond , x ) do { if ( cond ) { S2N_ERROR( x ); }} while (0)
+#define S2N_ERROR_UNLESS( cond , x ) S2N_ERROR_IF( !( cond ), x)
```

Comment:
Nit. I'm not sure if we really need `S2N_ERROR_UNLESS(cond, err)`. I think we should standardize on manually adding the `!` such as `S2N_ERROR_IF(!cond, err)`. It's shorter in terms of line length, and it makes it impossible to accidentally mix up the Macros if there's only one option.

Reply:
Done